### PR TITLE
Aerogear 8260

### PR DESCRIPTION
**Pull request by secondsun**
# What this does

Adds a bind and unbind options

# What might be missing

The current information in the binding is rather minimal, but that all can be added now that binding/unbinding works.

# How to test

You should be able to run `apb bundle prepare` then do a docker build and a docker push to whatever repository your broker is looking at.  You can then deploy the service and bind/unbind from MDC per normal.

### CODE REVIEW

**Comment by secondsun** [comment]
See https://www.youtube.com/watch?v=CctnZOd6Wxg&feature=youtu.be

File: roles/test-sync-app-apb/tasks/unbind.yml [hunk 1]
```diff
@@ -0,0 +1,6 @@
+---
+
+- name: "Test|Check that sync binding secret exists"
```
Comment (psturc) [inline roles/test-sync-app-apb/tasks/unbind.yml]: should be `... binding secret does not exist`?

File: roles/unbind-unifiedpush-apb/tasks/main.yml [hunk 2]
```diff
@@ -0,0 +1,2 @@
+- debug:
+    msg: "Sync does not need to release resources on unbind.."
```
Comment (psturc) [inline roles/unbind-unifiedpush-apb/tasks/main.yml]: Is this some role for testing?

Comment (secondsun) [reply roles/unbind-unifiedpush-apb/tasks/main.yml]: Unbind failed if I didn't have an unbind role.  I suppose I could have looked more into it, but it seemed better to put something that was of the "page intentionally left blank" ilk

File: roles/test-sync-app-apb/tasks/unbind.yml (hunk 1 above)
Comment (secondsun) [reply roles/test-sync-app-apb/tasks/unbind.yml]: +1

File: roles/bind-sync-app-apb/tasks/main.yml [hunk 3]
```diff
@@ -0,0 +1,45 @@
+---
+
+- name: "Retrieve route to sync"
+  shell: "oc get routes/sync-app -n '{{ namespace }}' -o jsonpath='{.spec.host}'"
```
Comment (wei-lee) [inline roles/bind-sync-app-apb/tasks/main.yml]: this probably will not work if there are multiple instances of the sync app

File: roles/unbind-unifiedpush-apb/tasks/main.yml [hunk 4]
```diff
@@ -0,0 +1,2 @@
+- debug:
```
Comment (wei-lee) [inline roles/unbind-unifiedpush-apb/tasks/main.yml]: the name of this file is wrong

File: roles/test-sync-app-apb/tasks/unbind.yml (hunk 1 above)
Comment (secondsun) [reply roles/test-sync-app-apb/tasks/unbind.yml]: @psturc Updated

File: roles/bind-sync-app-apb/tasks/main.yml (hunk 3 above)
Comment (secondsun) [reply roles/bind-sync-app-apb/tasks/main.yml]: @wei-lee That is part of a different JIRA. (See my multi binding branch)

File: roles/unbind-unifiedpush-apb/tasks/main.yml (hunk 4 above)
Comment (secondsun) [reply roles/unbind-unifiedpush-apb/tasks/main.yml]: What is the right name of the file?

Comment (secondsun) [reply roles/unbind-unifiedpush-apb/tasks/main.yml]: Oh I see now.  Forgot to git rm that file.

File: roles/unbind-unifiedpush-apb/tasks/main.yml (hunk 2 above)
Comment (secondsun) [reply roles/unbind-unifiedpush-apb/tasks/main.yml]: Ok I just double checked, I 100% committed this director by accident.

**Pull request merged**